lib/tasks/youth.rake

```diff
@@ -0,0 +1,286 @@
+        'White' => 'White / Caucasian',
+      }.invert[race_string]
+    end
+
+    def ethnicity(ethnicity_string)
+      {
+        0 => 'Non-Hispanic / Non-Latino',
+        1 => 'Hispanic / Latino',
+        8 => 'Unknown',
+        9 => 'Youth refused',
+        99 => 'Data not collected',
+      }.invert[ethnicity_string]
+    end
+
+    def gender(gender_string)
+      {
+        'Female' => 0,
+        'Male' => 1,
+        'Trans Male (MTF or Male to Female)' => 3,
+        'Trans Female (MTF or Male to Female)' => 2,
+        'Trans Male (FTM or Female to Male)' => 3,
+        'Trans Female (FTM or Female to Male)' => 2,
+        'Gender non-conforming (i.e. not exclusively male or female)' => 4,
+        'Unknown' => 8,
+        'Youth Refused' => 9,
+        'Data not collected' => 99,
+      }[gender_string]
+    end
+
+    data_source = GrdaWarehouse::DataSource.authoritative.youth.first_or_create do |ds|
+      ds.name = 'DIAL/SELF Youth'
+      ds.short_name = 'Youth'
+    end
+    GrdaWarehouse::Youth::YouthReferral.where(imported: true).destroy_all
+    GrdaWarehouse::Youth::YouthCaseManagement.where(imported: true).destroy_all
+    GrdaWarehouse::Youth::DirectFinancialAssistance.where(imported: true).destroy_all
+    GrdaWarehouse::YouthIntake::Base.where(imported: true).destroy_all
+    # Faker leaves a mess behind, so we'll clean this up for development runs
+    if Rails.env.development?
+      GrdaWarehouse::Hud::Client.where(data_source_id: data_source.id).destroy_all
+    end
+
+    xlsx = Roo::Spreadsheet.open(file_path)
+    sheet = xlsx.sheet(0)
+
+    client_names = {}
+    staff_names = {}
+    staff_email = {}
+    clients = {}
+    intakes_created = 0
+    intakes_failed = 0
+    # Process
+    # -1 destroy any previously imported intakes, referrals, manag
+    # 0. make data source DIAL/SELF Youth
+    # 0.5 sort by updated_at
+    # 1. Loop through xlsx
+    # 2. Pickup FirstName, LastName, DOB - find or create client in GrdaWarehouse::DataSource.authoritative.youth
+    # 2.5 match existing, not destination clients for creating intakes
+    # 3. Create one intake per client, potentially update if seen again
+    # 4. Create one case management note per row with youth_case_management == Yes
+    # 5. Create one referral per row with referrals == Yes
+    # 6. Create one direct financial assistance per row with financial_assistance_provided == Yes
+    sheet.each(headers) do |row|
+      next if row[:updated_at] == 'Timestamp'
+      # Basic anonymization
+      if Rails.env.development?
+        client_names[row[:PersonalID]] ||= Faker::Name.name
+        staff_names[row[:PersonalID]] ||= Faker::Name.name
+        staff_email[row[:email]] ||= Faker::Internet.email
+        row[:client_name] = client_names[row[:PersonalID]]
+        row[:staff_name] = staff_names[row[:PersonalID]]
+        row[:email] = staff_email[row[:email]]
+      end
```

Comment:
Should DOB be faked as well?

Reply:
We don't expect to use this again, and since these need to be youth, using real DOB seems safe.  This is just for local verification.